### PR TITLE
change dan to class

### PR DESCRIPTION
**Pull request by Ksoy**
single use
```
import dan
dan.register()
```

multiple use
```
from dan import Client
dan = Client()
dan.register()
```

### CODE REVIEW

File: iot/dan.py [hunk 1]
```diff
@@ -332,32 +332,31 @@ def push(self, idf, data, json_=False, block=False):
         return True
 
 
-_default_client = Client()
+class DAN():
 
+    def __init__(self):
+        self._default_client = Client()
```
Comment (iblis17) [inline iot/dan.py]: So, why do we need this class?
can those methods be merged into `Client` ?

Comment (Ksoy) [inline iot/dan.py]: Yeah, u r right.
it should be merged.

**Comment by Ksoy** [comment]
I think i am so stupid XD
Just import Client make everything well

**Comment by iblis17** [comment]
Ok, worth to document both usage on this file head?

File: iot/dan.py [hunk 2]
```diff
@@ -350,9 +353,7 @@ def register(url, on_signal, on_data,
 
 
 def deregister():
-    ret = _default_client.deregister()
-    _default_client.context.mqtt_client = None
-    return ret
+    return _default_client.deregister()
```
Comment (iblis17) [inline iot/dan.py]: Why this change?

Comment (Ksoy) [inline iot/dan.py]: move this line `_default_client.context.mqtt_client = None`

Comment (Ksoy) [inline iot/dan.py]: move this line `_default_client.context.mqtt_client = None`
to `Client.deregister()`

make sure multiple use still set it to None

File: iot/dan.py [hunk 3]
```diff
@@ -1,5 +1,25 @@
 '''
 This module wraps the mqtt API into IoTtalk client API
+
+If your process contain sigle Device,
+you can use
```
Comment (iblis17) [inline iot/dan.py]:
```diff
- you can use
+ you can use::
```

`::` denote a code block in RST

File: iot/dan.py [hunk 4]
```diff
@@ -299,7 +320,7 @@ def deregister(self):
                 raise RegistrationError(response.json()['reason'])
         except requests.exceptions.ConnectionError:
             raise RegistrationError('ConnectionError')
-
+        self.context.mqtt_client = None
```
Comment (iblis17) [inline iot/dan.py]: This line should be placed after `_disconn_lock.acquire()` to prevent some trivial race condition like:

1. `mqtt_client` set to `None`
2. the actual instance of `mqtt_client` being GCed.
3. the on_disconnect callback never be executed.
4. this lock block the program.

Comment (Ksoy) [inline iot/dan.py]: When is `on_disconnect` called?

Comment (Ksoy) [inline iot/dan.py]: Ok, i see
